rust/noosphere-ns/src/name_system.rs

```diff
@@ -39,24 +39,24 @@ lazy_static! {
 /// See <https://github.com/subconsciousnetwork/noosphere/blob/main/design/name-system.md> for
 /// the full Noosphere Name System spec.
 pub struct NameSystem {
-    pub(crate) dht: DHTNode,
-    /// Map of sphere DIDs to [NSRecord] hosted/propagated by this name system.
-    hosted_records: Mutex<HashMap<Did, NSRecord>>,
-    /// Map of resolved sphere DIDs to resolved [NSRecord].
-    resolved_records: Mutex<HashMap<Did, NSRecord>>,
+    pub(crate) dht: DhtNode,
+    /// Map of sphere DIDs to [NsRecord] hosted/propagated by this name system.
+    hosted_records: Mutex<HashMap<Did, NsRecord>>,
+    /// Map of resolved sphere DIDs to resolved [NsRecord].
+    resolved_records: Mutex<HashMap<Did, NsRecord>>,
 
     #[cfg(feature = "api_server")]
     api_server: Option<APIServer>,
 }
 
 impl NameSystem {
-    pub fn new<S: Storage + 'static, K: DHTKeyMaterial>(
+    pub fn new<K: DhtKeyMaterial, V: RecordValidator + 'static>(
         key_material: &K,
-        store: SphereDb<S>,
-        dht_config: DHTConfig,
+        dht_config: DhtConfig,
+        validator: Option<V>,
```

Comment:
I don't like a configurable/optional top-level validator (wanting a network of nodes to have relatively similar/consistent validators), but I don't have any other ideas in order to support this (and more pushing and pulling between static DhtConfig values and living instances. I'm glad you've spent some time in this module and the `Dht*` counterparts for future discussions on some of the cleanup challenges ahead!

Reply:
Pasting my related comment from Discord:

> I'm sympathetic to your position on configurable Validator. Ultimately, I don't think orb-ns will offer a configuration here. Nor would orb. And, it may prove to be an anti-pattern altogether.

And filed https://github.com/subconsciousnetwork/noosphere/issues/268